src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets

```diff
@@ -19,28 +23,44 @@
     </ImportGroup>
 
     <ImportGroup Condition="'$(TargetPlatformIdentifier)' == 'ios'">
-        <Import Project="Sdk.props" Sdk="Microsoft.NET.Runtime.RuntimeConfigParser.Task" />
+        <Import Project="Sdk.props" Sdk="Microsoft.NET.Runtime.MonoTargets.Sdk" />
+	<Import Project="%(ResolvedRuntimePack.PackageDirectory)\runtimes\$(RuntimeIdentifier)\build\component-manifest.props"/>
         <Import Project="Sdk.props" Sdk="Microsoft.NETCore.App.Runtime.AOT.Cross.ios-arm" />
         <Import Project="Sdk.props" Sdk="Microsoft.NETCore.App.Runtime.AOT.Cross.ios-arm64" />
         <Import Project="Sdk.props" Sdk="Microsoft.NETCore.App.Runtime.AOT.Cross.iossimulator" />
+	<Import Project="Sdk.targets" Sdk="Microsoft.NET.Runtime.MonoTargets.Sdk" />
     </ImportGroup>
 
     <ImportGroup Condition="'$(TargetPlatformIdentifier)' == 'maccatalyst'">
-        <Import Project="Sdk.props" Sdk="Microsoft.NET.Runtime.RuntimeConfigParser.Task" />
+        <Import Project="Sdk.props" Sdk="Microsoft.NET.Runtime.MonoTargets.Sdk" />
+	<Import Project="%(ResolvedRuntimePack.PackageDirectory)\runtimes\$(RuntimeIdentifier)\build\component-manifest.props"/>
         <Import Project="Sdk.props" Sdk="Microsoft.NETCore.App.Runtime.AOT.Cross.maccatalyst" />
+	<Import Project="Sdk.targets" Sdk="Microsoft.NET.Runtime.MonoTargets.Sdk" />
     </ImportGroup>
 
     <ImportGroup Condition="'$(TargetPlatformIdentifier)' == 'tvos'">
-        <Import Project="Sdk.props" Sdk="Microsoft.NET.Runtime.RuntimeConfigParser.Task" />
+        <Import Project="Sdk.props" Sdk="Microsoft.NET.Runtime.MonoTargets.Sdk" />
+	<Import Project="%(ResolvedRuntimePack.PackageDirectory)\runtimes\$(RuntimeIdentifier)\build\component-manifest.props"/>
         <Import Project="Sdk.props" Sdk="Microsoft.NETCore.App.Runtime.AOT.Cross.tvos-arm64" />
+	<Import Project="Sdk.targets" Sdk="Microsoft.NET.Runtime.MonoTargets.Sdk" />
+    </ImportGroup>
+
+    <ImportGroup Condition="'$(TargetPlatformIdentifier)' == 'tvossimulator'">
+        <Import Project="Sdk.props" Sdk="Microsoft.NET.Runtime.MonoTargets.Sdk" />
+	<Import Project="%(ResolvedRuntimePack.PackageDirectory)\runtimes\$(RuntimeIdentifier)\build\component-manifest.props"/>
         <Import Project="Sdk.props" Sdk="Microsoft.NETCore.App.Runtime.AOT.Cross.tvossimulator" />
+	<Import Project="Sdk.targets" Sdk="Microsoft.NET.Runtime.MonoTargets.Sdk" />
     </ImportGroup>
 
     <ImportGroup Condition="'$(RuntimeIdentifier)' == 'browser-wasm' and '$(UsingBrowserRuntimeWorkload)' == 'true'">
+        <Import Project="Sdk.props" Sdk="Microsoft.NET.Runtime.MonoTargets.Sdk" />
+	<Import Project="%(ResolvedRuntimePack.PackageDirectory)\runtimes\$(RuntimeIdentifier)\build\component-manifest.props"/>
```

Comment:
Does `%(ResolvedRuntimePack.PackageDirectory)` work here? IIUC, batching shouldn't work here at all. And AFAIK, `@(ResolvedRuntimePack)` is set by `ResolveFrameworkReferences`. But I could be wrong. Does it work for you?

Reply:
I wasn't 100% sure either and I think we need to test to find out

Reply:
I tested it in a target at which point it would be resolved I think we may need a couple different plan.

Reply:
Pinged some MSBuild folks.  One thing I coudl do is use the `<MSBuild/>` task in my MonoTargets target to read the manifest, but since there's a couple different things in there, it's actually a little bit easier/faster/cleaner to write a little Task that will read a component manifest json blob and get everything from there.  So I'll hack that up (probably tomorrow)

Reply:
That task will be useful for wasm build targets too!